lib/slack/post.rb

```diff
@@ -64,18 +61,18 @@ def self.post_url
 		
 		LegacyConfigParams = [:subdomain,:token].freeze
 		
-		def self.configured?(needs_channel=true)
-			return false if needs_channel and !config[:channel]
 
-			# we need _either_ a webhook url or all LegacyConfigParams
+		# we need _either_ a webhook url or all LegacyConfigParams
+		def self.configured?
 			return true if config[:webhook_url]
 			LegacyConfigParams.all? do |parm|
 				config[parm]
+				config[:channel] = '#general' unless config[:channel]
```

Comment:
I don't think this line is doing what you expect. The block returns `true` if `config[foo]` is defined for all the `foo`s in `LegacyConfigParams`. You're setting a value for an unrelated config param on each iteration of the block.